src/lib/parameters/parameters.cpp

```diff
@@ -664,6 +668,32 @@ param_control_autosave(bool enable)
 	param_unlock_writer();
 }
 
+static void notify_worker(void *arg)
+{
+	param_notify_changes();
+	notify_delay_count.store(0);
+}
+
+static void schedule_notify_changes()
+{
+#if defined(ENABLE_LOCKSTEP_SCHEDULER) || defined(CONFIG_ARCH_BOARD_PX4_SITL)
+	// on SITL call notify_worker directly to avoid issues with lockstep and SITL tests
+	(void)notify_work;
+	notify_worker(nullptr);
+#else
+
+	if (notify_delay_count.fetch_add(1) < 100) {
+		// delay notification by 20 ms, but no more than 100 times
+		work_queue(LPWORK, &notify_work, &notify_worker, nullptr, USEC2TICK(20'000));
+
+	} else {
+		// otherwise schedule immediately
+		work_queue(LPWORK, &notify_work, &notify_worker, nullptr, 0);
+	}
```

Comment:
Once one write (from count >=100) is scheduled  but not executed, this should not reschedule on subsequent call. You could use a int for the type and Here set it to  -1, check it on entry and return if negative and bail.

Reply:
Updated.